src/amp-story-player/amp-story-player-impl.js

```diff
@@ -1264,6 +1263,39 @@ export class AmpStoryPlayer {
     );
   }
 
+  /**
+   * Returns the storyIdx of a story given a URL.
+   * @param {string} storyUrl
+   * @return {number}
+   * @private
+   */
+  getStoryIdxFromUrl_(storyUrl) {
+    // TODO(enriqe): sanitize URLs for matching.
+    const storyIdx = storyUrl
+      ? findIndex(this.stories_, ({href}) => href === storyUrl)
+      : this.currentIdx_;
+
+    // TODO(#28987): add() stories when show() calls this and story isn't added
+    // yet.
+    if (!this.stories_[storyIdx]) {
+      throw new Error(`Story URL not found in the player: ${storyUrl}`);
+    }
+
+    return storyIdx;
+  }
+
+  /**
+   * Rewinds the given story.
+   * @param {string} storyUrl
+   */
+  rewind(storyUrl) {
+    const story = this.stories_[this.getStoryIdxFromUrl_(storyUrl)];
+
+    story.messagingPromise.then((messaging) =>
+      messaging.sendRequest('selectPage', {'rewind': true})
```

Comment:
Re: introducing a new message, I really think a `rewind` API makes sense as it does more than just selecting a page, it's not equivalent to navigating to page 0